corehq/middleware.py

```diff
@@ -116,10 +116,10 @@ class TimeoutMiddleware(MiddlewareMixin):
     @classmethod
     def update_secure_session(cls, session, is_secure, domain=None):
         session['secure_session'] = is_secure
-        session['last_request'] = json_format_datetime(datetime.datetime.utcnow())
         timeout = cls._get_timeout(session, is_secure, domain)
         session['secure_session_timeout'] = timeout
         session.set_expiry(timeout * 60)
+        session['session_expiry'] = session.get_expiry_date()
```

Comment:
Curious why this needs to be stored in the session? Would it work just as well to call `session.get_expiry_date()` in `ping_response`?

Reply:
Hmm...maybe it doesn't need to be stored here. I'll keep that in mind (I'm still working on this, adding in the logic we discussed to use the minimum timeout length of all domains where the user is a member).

Reply:
After working with this code some more, I'm inclined to keep this as-is. The javascript now uses `session_expiry`, and storing it in the session makes it easily available on page load.